API/Startup.cs

```diff
@@ -53,6 +55,30 @@ public void ConfigureServices(IServiceCollection services)
                 });
             });
 
+            services.AddIdentity<UserDTO, IdentityRole>()
+                .AddEntityFrameworkStores<ApiContext>();
+
+            services.Configure<IdentityOptions>(options =>
+            {
+                // Password settings.
+                options.Password.RequireDigit = true;
+                options.Password.RequireLowercase = true;
+                options.Password.RequireNonAlphanumeric = true;
+                options.Password.RequireUppercase = true;
+                options.Password.RequiredLength = 6;
+                options.Password.RequiredUniqueChars = 1;
+
+                // Lockout settings.
+                options.Lockout.DefaultLockoutTimeSpan = TimeSpan.FromMinutes(5);
+                options.Lockout.MaxFailedAccessAttempts = 5;
+                options.Lockout.AllowedForNewUsers = true;
+
+                // User settings.
+                options.User.AllowedUserNameCharacters =
+                "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_";
```

Comment:
Is it really wise to hardcode a string like this? Maybe it is possible to use regex here?

Reply:
See https://github.com/dotnet/aspnetcore/issues/14511